packages/arb-token-bridge-ui/src/hooks/useSwitchNetworkWithConfig.tsx

```diff
@@ -0,0 +1,27 @@
+import { useSwitchNetwork } from 'wagmi'
+
+import {
+  handleSwitchNetworkError,
+  handleSwitchNetworkOnMutate
+} from '../util/networks'
+
+type SwitchNetworkConfig = {
+  isSwitchingNetworkBeforeTx?: boolean
+}
+
+export function useSwitchNetworkWithConfig({
+  isSwitchingNetworkBeforeTx = false
+}: SwitchNetworkConfig = {}) {
+  const config = {
+    throwForSwitchChainNotSupported: true,
+    onMutate: () => handleSwitchNetworkOnMutate({ isSwitchingNetworkBeforeTx }),
+    onError: handleSwitchNetworkError
+  }
+
+  const { switchNetwork, switchNetworkAsync } = useSwitchNetwork(config)
```

Comment:
I'm thinking we should just return everything from `useSwitchNetwork`, there's useful stuff in there